gvm/protocols/gmpv7/types.py

```diff
@@ -602,6 +607,12 @@ def get_scanner_type_from_string(
     ):
         return ScannerType.GMP_SCANNER_TYPE
 
+    if (
+        scanner_type == ScannerType.GREENBONE_SENSOR_SCANNER_TYPE.value
+        or scanner_type == 'greenbone'
+    ):
+        return ScannerType.GREENBONE_SENSOR_SCANNER_TYPE
+
```

Comment:
Same here. The function needs to be copied to gmpv9 and adjusted there only. GMP protocols before version 9 don't have this scanner type.